lib/models/service.ts

```diff
@@ -67,22 +67,20 @@ export interface IRetrievedService
  * 
  * @param authorizedRecipients 
  */
-export function toAuthorizedRecipientsSet(
+export function toAuthorizedRecipients(
   authorizedRecipients: ReadonlyArray<string> | ReadonlySet<string>
 ): ReadonlySet<FiscalCode> {
-  return authorizedRecipients && authorizedRecipients instanceof Array
-    ? new Set(authorizedRecipients.filter(isFiscalCode))
-    : authorizedRecipients instanceof Set
-      ? new Set(Array.from(authorizedRecipients).filter(isFiscalCode))
-      : new Set();
+  return new Set(
+    Array.from(authorizedRecipients ? authorizedRecipients : []).filter(
```

Comment:
the type of `authorizedRecipients` should reflect that it may be `undefined`: `ReadonlyArray<string> | ReadonlySet<string> | undefined`, unless you call `toAuthorizedRecipients` only when `authorizedRecipients` is defined

Reply:
I've added undefined to accepted types